src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncWithReferencedProductsInAnyOrderIT.java

```diff
@@ -112,6 +112,7 @@ private ProductSyncOptions buildSyncOptions() {
                                         .errorCallback((syncException, draft, product, updateActions)
                                             -> collectErrors(syncException.getMessage(), syncException))
                                         .beforeUpdateCallback(this::collectActions)
+                                        .batchSize(1)
```

Comment:
why we need to make batch size to 1? how does it relate to productWith500ProductReferences test case?

Reply:
caches are based on the batch, so it's ensuring the keyToId is considering other batches too, before it was not.